version: '3.8'

services:
  postgres:
    image: postgres:latest
    container_name: my_postgres
    ports:
      - "5432:5432"
    environment:
      POSTGRES_DB: refood_db
      POSTGRES_USER: refood
      POSTGRES_PASSWORD: refood
    volumes:
      - postgres-data:/var/lib/postgresql/data
    networks:
      - shared-network

  api:
    build: ./api
    ports:
      - "8080:80"
    command: bash -c "python seed_db.py && uvicorn app.main:app --host 0.0.0.0 --port 80"
    volumes:
      - ./api/app:/code/app
      - ./api/seed_db.py:/code/seed_db.py
    depends_on:
      - postgres
    networks:
      - shared-network
  
  frontend:
    build: ./frontend
    command: nginx -g "daemon off;"
    ports:
      - "8090:80"

  elasticsearch:
    container_name: elasticsearch
    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.0
    environment:
      - discovery.type=single-node
    volumes:
      - elasticsearch-data:/usr/share/elasticsearch/data
    ports:
      - "9200:9200"
      - "9300:9300"
    networks:
      - shared-network

  rabbitmq:
    container_name: rabbitmq
    image: rabbitmq:3-management
    ports:
      - "15672:15672"
      - "5672:5672"
    networks:
      - shared-network
    healthcheck:
      test: "exit 0"

  fulltext:
    build: ./fulltext
    container_name: fulltext
    volumes:
      - ./fulltext/app:/code
    working_dir: /code
    command: python3 main.py
    networks:
        - shared-network
    depends_on:
        - elasticsearch
        - rabbitmq

volumes:
  postgres-data:
  elasticsearch-data:

networks:
  shared-network:
